packages/frontend-2/components/server-management/Table.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="text-foreground">
+    <div class="w-full text-sm overflow-x-auto overflow-y-visible simple-scrollbar">
+      <div
+        class="grid z-10 grid-cols-12 items-center gap-6 font-semibold bg-foundation rounded-t-lg w-full border-b border-outline-3 pb-2 pt-4 px-4 min-w-[900px]"
+        :style="{ paddingRight: paddingRightStyle }"
+      >
+        <div
+          v-for="header in headers"
+          :key="header.id"
+          :class="columnClasses[header.id]"
+          class="capitalize"
+        >
+          {{ header.title }}
+        </div>
+      </div>
+      <div
+        class="divide-y divide-outline-3 h-full overflow-visible"
+        :class="{ 'pb-32': overflowCells }"
+      >
+        <div
+          v-for="item in items"
+          :key="item.id"
+          class="relative grid grid-cols-12 items-center gap-6 px-4 py-1 min-w-[900px] bg-white"
+          :style="{ paddingRight: paddingRightStyle }"
+          :class="{ 'cursor-pointer hover:bg-primary-muted': !!props.onRowClick }"
+          tabindex="0"
+          @click="handleRowClick(item)"
+          @keydown.enter.space="handleRowClick(item)"
+        >
+          <template v-for="(column, colIndex) in headers" :key="column.id">
+            <div :class="getClasses(column.id, colIndex)" tabindex="0">
+              <slot :name="column.id" :item="item">
+                <div class="text-gray-900 font-medium order-1">
+                  {{ (item as any)[column.id] }}
+                </div>
+              </slot>
+            </div>
+          </template>
+          <div class="absolute right-0 flex items-center p-0 pr-0.5">
+            <div v-for="(button, btnIndex) in buttons" :key="btnIndex" class="p-1">
```

Comment:
button.label is probably a better key, there's never gonna be multiple buttons with the same label